Lib/typing.py

```diff
@@ -981,7 +981,7 @@ def __eq__(self, other):
         return set(_value_and_type_iter(self.__args__)) == set(_value_and_type_iter(other.__args__))
 
     def __hash__(self):
-        return hash(tuple(_value_and_type_iter(self.__args__)))
+        return hash(frozenset(_value_and_type_iter(self.__args__)))
```

Comment:
Given that the sets are all thrown away after a single use, I think we should use `set()` everywhere or `frozenset()` everywhere. I vote for using `set()`.

Reply:
`set()` seems to be unhashable:
```python
>>> hash({1, 2, 3})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type: 'set'
```

Reply:
Shows how much I know. :-)

Reply:
I was curious about how `Union.__hash__` is implemented and it follows the same pattern:
https://github.com/python/cpython/blob/fa2eee975dbf7d2728021ef9d97328bbe88351cf/Lib/typing.py#L950-L951